go/interpreter/lfvm/instructions.go

```diff
@@ -1193,42 +1080,56 @@ func opCallCode(c *context) {
 	if err != nil {
 		return
 	}
+
 	needed_memory_size := arg_memory_size
 	if ret_memory_size > arg_memory_size {
 		needed_memory_size = ret_memory_size
 	}
-	base_gas := c.memory.ExpansionCosts(needed_memory_size)
-	if base_gas < 0 || c.gas < base_gas {
+
+	baseGas := c.memory.ExpansionCosts(needed_memory_size)
+	checkGas := func(cost tosca.Gas) bool {
+		if cost < 0 || c.gas < cost {
+			// c.status = OUT_OF_GAS
+			return false
+		}
+		return true
+	}
+	if !checkGas(baseGas) {
 		c.status = OUT_OF_GAS
 		return
 	}
 
-	// We need to check the existence of the target account before removing
-	// the gas price for the other cost factors to make sure that the read
-	// in the state DB is always happening. This is the current EVM behavior,
-	// and not doing it would be identified by the replay tool as an error.
-	toAddr := tosca.Address(addr.Bytes20())
-
+	// for static and delegate calls, the following value checks will always be zero.
 	// Charge for transferring value to a new address
 	if !value.IsZero() {
-		base_gas += tosca.Gas(params.CallValueTransferGas)
+		baseGas += tosca.Gas(params.CallValueTransferGas)
+	}
+	if !checkGas(baseGas) {
+		c.status = OUT_OF_GAS
+		return
 	}
 
-	cost := callGas(c.gas, base_gas, provided_gas)
+	// if evm.chainRules.IsEIP158 according to GETH it is EIP158 since 2016
+	// !!!! but need to touch stateDB for the address to have it in the substate record key/value
```

Comment:
nit: I'd say this comment needs improvement ...

Reply:
this comment is prior to this PR and was not modified, just relocated. That said, I also found it strange, so I rephrased it.